website_sale_checkout_store/controllers/main.py

```diff
@@ -53,3 +53,13 @@ def payment_get_status(self, sale_order_id, **post):
             return {'recall': False, 'message':''}
         else:
             return super(website_sale, self).payment_get_status(sale_order_id, **post)
+
+    @http.route(['/shop/payment/transaction/<int:acquirer_id>'], type='json', auth="public", website=True)
+    def payment_transaction(self, acquirer_id):
+        cr, uid, context = request.cr, request.uid, request.context
+        order = request.website.sale_get_order(context=context)
+        if 'noship' not in order.buy_way:
+            return super(website_sale, self).payment_transaction(acquirer_id)
+        else:
+            order.partner_id.country_id = request.env['res.country'].browse(1)
```

Comment:
make the same country as admin has

            order.partner_id.country_id = request.env.sudo().user.partner_id.country_id.id